File: Client/Gamify.Client.Net/Gamify.Client.Net/Services/GameService.cs
using Gamify.Client.Net.Client;
using Gamify.Client.Net.Contracts.Notifications;
using Gamify.Client.Net.Contracts.Requests;

namespace Gamify.Client.Net.Services
{
    public class GameService<TRequest, UNotification> : GameListener<UNotification>, IGameService<TRequest, UNotification>
        where TRequest : IRequestObject
        where UNotification : INotificationObject
    {
        private readonly GameRequestType requestType;
        private readonly ISerializer serializer;

        public GameService(GameRequestType requestType, GameNotificationType notificationType, IGameClient gameClient)
            : base(notificationType, gameClient)
        {
            this.requestType = requestType;
            this.serializer = new JsonSerializer();
        }

        public void Send(TRequest request)
        {
            var serializedGameRequest = this.serializer.Serialize(request);
            var gameRequest = new GameRequest
            {
                Type = (int)this.requestType,
                SerializedRequestObject = serializedGameRequest
            };

            this.gameClient.Send(gameRequest);
        }
    }
}

File: Client/Java/gamify-java-client/src/main/java/ar/com/mdf/gamify/services/GameService.java
package ar.com.mdf.gamify.services;

import ar.com.mdf.gamify.client.IGamifyClient;
import ar.com.mdf.gamify.enums.NotificationType;
import ar.com.mdf.gamify.enums.RequestType;
import ar.com.mdf.gamify.objects.notification.GameNotification;
import ar.com.mdf.gamify.objects.notification.INotificationObject;
import ar.com.mdf.gamify.objects.request.GameRequest;
import ar.com.mdf.gamify.objects.request.IRequestObject;
import ar.com.mdf.gamify.observers.IGameMessageObserver;
import ar.com.mdf.gamify.observers.IGameObserver;
import ar.com.mdf.gamify.parsers.GameParser;
import ar.com.mdf.gamify.parsers.IGameParser;

public class GameService <T extends IRequestObject, U extends INotificationObject > extends GameListener<U> implements IGameService<T, U>, IGameMessageObserver {
	
	private RequestType requestType;
	private IGamifyClient gameClient;
	private IGameParser<T> requestParser;
	
	public GameService(RequestType rt, NotificationType nt, IGamifyClient client, Class<U> notificationClazz) {
		super(nt, notificationClazz);
		this.requestType = rt;
		this.gameClient = client;
		this.requestParser = new GameParser<T>();
		this.gameClient.suscribe(this);
	}

	protected IGameParser<T> getRequestParser() {
		return requestParser;
	}
	
	
	public void send(T request) {
		GameRequest gr = new GameRequest();
		gr.setType(this.requestType.getType());
		String serializedReuqest = this.getRequestParser().parseToString(request);
		gr.setSerializedRequestObject(serializedReuqest);
		this.gameClient.send(gr);
	}
	
		
	public void disConnect() {
		this.gameClient.close();
	}

	
}
